app/src/main/java/org/oppia/android/app/home/topiclist/TopicSummaryViewModel.kt

```diff
@@ -24,12 +33,118 @@ class TopicSummaryViewModel(
   val backgroundColor: Int = retrieveBackgroundColor()
   @ColorInt
   val darkerBackgroundOverlayColor: Int = computeDarkerBackgroundColor()
+  private val marginTopBottom = (activity as Context).resources
+      .getDimensionPixelSize(R.dimen.topic_list_item_margin_top_bottom)
+  private val marginMax = (activity as Context).resources.getDimensionPixelSize(R.dimen.home_margin_max)
+  private val marginMin = (activity as Context).resources.getDimensionPixelSize(R.dimen.home_margin_min)
+  private var position = -1
+  private var spanCount = activity.resources.getInteger(R.integer.home_span_count)
 
   /** Callback from data-binding for when the summary tile is clicked. */
   fun clickOnSummaryTile() {
     topicSummaryClickListener.onTopicSummaryClicked(topicSummary)
   }
 
+  fun setSpanCount(newSpan: Int) {
+    this.spanCount = newSpan
+  }
+
+  fun setPosition(newPosition: Int) {
+    this.position = newPosition
+  }
+
+  @BindingAdapter("android:layout_marginTop")
+  fun setLayoutMarginTop(view: View, viewModel: TopicSummaryViewModel) {
```

Comment:
This should be returning an int--at the moment you're trying to assign a Unit (default return type for Kotlin functions) to an integer field (layout_marginTop). I think the layout is failing to compile (& therefore generating the binding impl) which is leading to downstream failures in the build pipeline.

Reply:
Done